content/api/commands/selectfile.md

```diff
@@ -0,0 +1,295 @@
+initially exist until the file exists or the command times out.
+
+```javascript
+// will fail after the defaultCommandTimeout is reached
+cy.get('input[type=file]').selectFile('does-not-exist.yaml')
+```
+
+### Actionability
+
+#### The element must first reach actionability
+
+`.selectFile()` is an "action command" that follows all the rules of
+[Actionability](/guides/core-concepts/interacting-with-elements).
+
+## Rules
+
+### Requirements [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Chains-of-Commands)
+
+- `.selectFile()` requires being chained off a command that yields DOM
+  element(s). With the `input` action (default), it further requires a single
+  `input` element with `type="file"`, or a `label` element attached to one.
+- If given a path, `.selectFile()` requires the file must exist.
+- If given an alias, `.selectFile()` requires that the subject of the alias must
+  not be `null` or `undefined`.
+
+### Assertions [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Assertions)
+
+- `.selectFile()` will automatically wait for the element to reach an
+  [actionable state](/guides/core-concepts/interacting-with-elements).
+
+### Timeouts [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Timeouts)
+
+- `.selectFile()` can time out waiting for the element to reach an
+  [actionable state](/guides/core-concepts/interacting-with-elements).
+- `.selectFile()` can time out waiting for a file to exist on disk or for an
+  alias to resolve.
+
+## Command Log
+
+**_Select file for input_**
+
+```javascript
+cy.get('.file-input').selectFile(Cypress.Buffer.from('Hello world'))
+```
+
+<!--
+Below screenshots generated from this HTML:
+
+<form>
+  <input id="basic" type="file" className="file-input" />
+</form>
+
+And this test:
+
+it('should attach a file', () => {
+  cy.get('.file-input').selectFile(Cypress.Buffer.from('Hello world'))
+})
+-->
+
+The commands above will display in the Command Log as:
+
+<DocsImage src="/img/api/selectfile/selectfile-command-log.png" alt="Command log for selectFile" ></DocsImage>
+
+When clicking on `selectFile` within the command log, the console outputs the
+following:
+
+<DocsImage src="/img/api/selectfile/selectfile-console.png" alt="console.log for selectFile" ></DocsImage>
+
+## History
+
+| Version                                     | Changes                       |
+| ------------------------------------------- | ----------------------------- |
+| [9.3.0](/guides/references/changelog#9.3.0) | `.selectFile()` command added |
+
+### Community Recognition
+
+The `.selectFile()` command draws heavy inspiration from the now-deprecated
+Cypress File Upload plugin. It was made possible by
+[@abramenal](https://github.com/abramenal) and contributors to the
+[cypress-file-upload](https://github.com/abramenal/cypress-file-upload#contributors)
+repository.
+
```

Comment:
@cowboy / @BlueWinds  Was there intentions of linking to the `cypress-file-uploaded` migration guide here?

Reply:
I wasn't planning on it. IMO the docs are a bit more 'permanent', while the migration guide is intended to fall off in visibility (get buried as we release more versions).

I'm not against adding a link if people feel it would be helpful, but not sure it's necessary here (since there are other links to it scattered around).

Reply:
I'll have to look and see. It shouldn't hold up the release, however. We can always add it afterwards.